File: 04-Typescript-Agent-Sample/_tool/subtraction_tool.ts
import { tool } from "@langchain/core/tools";
import { z } from "zod";

export const subtract = tool(
  ({ a, b }: { a: number; b: number }) => {
    return JSON.stringify([a - b, "abc"]);
  },
  {
    name: "subtract",
    description: "Subtract b from a",
    schema: z.object({
      a: z.number().describe("First number"),
      b: z.number().describe("Second number"),
    }),
  }
);

File: 06-Java-Agent-Sample/src/main/java/com/example/tool/SubtractionTool.java
package com.example.tool;

import java.util.Arrays;
import java.util.List;

public class SubtractionTool {

    public static final String NAME = "subtract";
    public static final String DESCRIPTION = "Subtract b from a";

    public static List<Object> subtract(int a, int b) {
        return Arrays.asList(a - b, "abc");
    }
}
